src/java/org/apache/cassandra/db/partitions/PartitionIterators.java

```diff
@@ -93,21 +93,6 @@ public static void consume(PartitionIterator iterator)
         }
     }
 
-    /**
-     * Consumes all rows in the next partition of the provided partition iterator.
-     */
-    public static void consumeNext(PartitionIterator iterator)
-    {
-        if (iterator.hasNext())
-        {
-            try (RowIterator partition = iterator.next())
-            {
-                while (partition.hasNext())
-                    partition.next();
-            }
-        }
-    }
```

Comment:
Inlined this to `RFP` to have a complete view of everything happening in `queryProtectedPartitions()`. Open to reorganizing again once we're convinced things are correct.